homeworks/roman.lytvynenko_Lytvynenko18/GamesProject/Scripts/script.js

```diff
@@ -0,0 +1,88 @@
+const EMAIL_INPUT_ID = 'email';
+const PASSWORD_INPUT_ID = 'password';
+const NOT_A_ROBOT_CHECKBOX_ID = 'checkbox';
+const SUBMIT_BUTTON_ID = 'submitButton';
+const ERRORS_CONTAINER_ID = 'errors-container';
+const RESULT_PAGE_PATH = './success.html';
+
+const submitButton = document.getElementById(SUBMIT_BUTTON_ID);
+
+/**
+ * Return input value by id.
+ * @param {string} elementId
+ * @return {string|boolean} input value
+ */
+function getValueById(elementId) {
+    const element = document.getElementById(elementId);
+    const type = element.getAttribute('type');
+    return type === 'checkbox' ? element.checked : element.value;
+}
+
+/**
+ * Add errors to errors container.
+ * @param {Object} inputData in format like: { [input_id]: error_text, ... }
+ */
+function setErrors(inputData) {
+    const errorContainerElement = document.getElementById(ERRORS_CONTAINER_ID);
+    Object.values(inputData).forEach((error) => {
+        const errorElement = document.createElement('p');
+        errorElement.classList.add('error');
+        errorElement.textContent = error;
+        errorContainerElement.appendChild(errorElement);
+    });
+}
+
+/**
+ * Delete all errors from errors container.
+ */
+function deleteErrors() {
+    const errorContainerElement = document.getElementById(ERRORS_CONTAINER_ID);
+    errorContainerElement.replaceChildren();
+}
+
+/**
+ * Goes to the page with the result.
+ */
+function navigateToResultPage() {
+    window.location.href = RESULT_PAGE_PATH;
+}
+
+function isEmail(email) {
+    return /\S+@\S+\.\S+/.test(email);
+}
+
+function deleteSpaces(email) {
+    return email.replace(/\s/g, '');
+}
+
+function passwordLength(password) {
+    return password.length >= 8 && password.length <= 22;
+}
+
+function validateForm() {
+    deleteErrors();
+    const email = getValueById(EMAIL_INPUT_ID);
+    const password = getValueById(PASSWORD_INPUT_ID);
+    const checkbox = getValueById(NOT_A_ROBOT_CHECKBOX_ID);
+    const validEmail = deleteSpaces(email);
+
+    if (!isEmail(validEmail)) {
+        setErrors({ ERRORS_CONTAINER_ID: 'Email invalid!' });
+    }
+
+    if (!passwordLength(password)) {
+        setErrors({ ERRORS_CONTAINER_ID: 'Password should from 8 to 12 chars!' });
+    }
+
+    if (!checkbox) {
+        setErrors({ ERRORS_CONTAINER_ID: 'Check the box that you are not a robot!' });
+    }
+
+    if (isEmail(validEmail) && passwordLength(password) && checkbox) {
```

Comment:
You make the same operations here for the second time.
To not violate the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) you can refactor you code:

```
declare an empty errors object
if (email is not valid) {
errors[email_id] = 'text of error'
}
... // same for other errors

if (error object has any key) {
  setErrors(errors object)
}

```

Reply:
I don't get it what U mean?
I wrote sequential checks for each field. If all data are correct, redirect to the "Success" page.
Could U please give me some details here,?

Reply:
I mean, you call `isEmail` and `passwordLength(password)` twice. And you call `setErrors` 3 times for each error. 
With this aproach, which I've described [here](https://github.com/MastersAcademy/fe-course-basic-2023/pull/314#discussion_r1393869160) you would not repeat yourself and call `isEmail` and `passwordLength(password)` only 1 time. Same for `setErrors` - you would call it once with whole bunch of errors

Reply:
But that is ok for now, your code is working properly, and my comments are related to refactoring and optimisation only. So I will merge it.